_posts/2025-09-30-Introducing-Derived-Source.markdown

```diff
@@ -0,0 +1,161 @@
+
+The `_source` field is not just used for document retrievals during search operations, but also for operations like updates, reindex operations, scripted updates, and recovery operations. If you disable `_source`, these operations become unavailable, which prevents data recovery and is therefore not recommended.
+
+OpenSearch 2.9.0 introduced [ZSTD compression](https://docs.opensearch.org/latest/im-plugin/index-codecs/), offering a high compression ratio at a fast compression speed. In the same experiment measuring storage footprint, enabling ZSTD compression reduced the stored field size to 216 GB---about a 46% reduction. However, even with this reduction, stored fields still occupy a significant amount of storage.
+
+
+## What is derived source?
+
+If your use case requires aggregations such as `min`, `max`, `avg`, `sum`, or `terms`, but not the matched documents themselves, storing the full data provides diminishing returns because the actual documents are unnecessary—the aggregations alone are sufficient.
+
+Starting with OpenSearch 3.2.0, you can use _derived source_ for such use cases to optimize the storage. Derived source mode modifies index behavior to exclude the `_source` field during ingestion, preventing data duplication and reducing storage requirements. These documents are retrieved dynamically using different forms of field storage (such as `doc_values` or stored fields) on demand. This approach preserves search functionality and supports operations that rely on `_source` data—such as reindexing, updates, scripted updates, and recovery, without actually storing the `_source` field.
+
+With this modified document retrieval behavior, during the fetch phase of a search query, OpenSearch retrieves each field’s value using formats such as `doc_values` and stored fields, then combines the results to produce the final document, as shown in the following image.
+
+![Derived_Source_Generation](/assets/media/blog-images/2025-09-30-Introducing-Derived-Source/derived-source-generation.png){:class="img-centered"}
+
+When configuring an index with derived source, OpenSearch validates that all fields are of supported types. When documents are accessed through operations such as search, update, or recovery, the derived source reconstructs each field’s value from either `doc_values` or stored fields as defined in the index mapping. Because this requires reading data from each field’s disk location rather than a single `_source` fetch, you may notice some latency degradation when retrieving large numbers of documents.
+
+
+## How to configure derived source?
+
+You can configure derived source at the index level. Because it changes the default behavior of storing the original `_source`, this setting can’t be updated after the index is created. This restriction prevents mixed behavior between the original stored source and the dynamically generated source (which can appear similar to the original source in output).
```

Comment:
```suggestion
You can configure derived source at the index level. Because it changes the default behavior of storing the original `_source`, this setting can't be updated after the index is created. This restriction prevents mixed behavior between the original stored source and the dynamically generated source (which can appear similar to the original source in output).
```